sdk/test-utils/recorder/test/browser/recorder.spec.ts

```diff
@@ -0,0 +1,296 @@
+import { RecorderEnvironmentSetup } from "../../src/utils";
+import chai from "chai";
+import { record } from "../../src";
+import xhrMock from "xhr-mock";
+const { expect } = chai;
+
+const expectedHttpResponse = "Hello World!";
+
+async function helloWorldRequest(): Promise<string> {
+  return new Promise((resolve) => {
+    function reqListener(this: any) {
+      resolve(this.responseText);
+    }
+
+    const req = new XMLHttpRequest();
+    req.addEventListener("load", reqListener);
+    req.open("GET", (window as any).__env__.PATH!);
+    req.send();
+  });
+}
+
+const recorderEnvSetup: RecorderEnvironmentSetup = {
+  replaceableVariables: {
+    PATH: "/replaced"
+  },
+  customizationsOnRecordings: [],
+  queryParametersToSkip: []
+};
+
+/**
+ * A function that generates another function with a predictable shape, even after compiling it to the browser.
+ */
+const getNoOpFunction = () => eval("() => {}");
+
+/**
+ * Another function that generates another function with a predictable shape, even after compiling it to the browser.
+ */
+const getAnotherNoOpFunction = () => eval("x => x");
+
+describe("The recorder's public API, on a browser", () => {
+  afterEach(() => {
+    delete (window as any).__env__.TEST_MODE;
+    delete (window as any).__env__.PATH;
+  });
+
+  it("should record a simple test", async function() {
+    // Setting up the record mode, and the PATH environment variable.
+    (window as any).__env__.TEST_MODE = "record";
+    (window as any).__env__.PATH = "/to/replace";
+
+    // We can't use Nise's FakeServer since the recorder ends up sending the request through the original XHR anyway.
+    xhrMock.setup();
+    xhrMock.get("/to/replace", {
+      status: 200,
+      body: expectedHttpResponse
+    });
+
+    // Before starting the recorder, we need to make a copy of the original XHR object, so that we can
+    // restore it before doing the recorder do its magic, in order for the HTTP request to hit the xhr-mock instance.
+    const originalXHR = XMLHttpRequest;
+
+    // The recorder outputs files into the console,
+    // so we need to mock the console.log function to capture and test the recorder output.
+    const originalConsoleLog = console.log;
+    const savedConsoleLogParams: any[] = [];
+    console.log = (...params: any) => {
+      savedConsoleLogParams.push(params);
+    };
+
+    // The recorder should start in the beforeEach call.
+    // We have to do this to emulate that.
+    (this as any).currentTest = {
```

Comment:
this feels weird, since we're mutating the context of the executing test, right? Can't we just set a mocked `this` value when we call record instead?

Reply:
For sure!

Reply:
I made a mocked this, but I had to use "any", because Mocha's Context is a class, and I wasn't able to make rollup work with Mocha as a dependency for both Node and the Browser during considerable time.

Reply:
I'll write up a new type to aid in this.

Reply:
Done!